### PR TITLE
@1aurabrown => Partner redirect

### PR DESCRIPTION
**Pull request by kanaabe**
- Adds a redirect for partner-written articles
- Adds embedded articles to `partner` subapp

Closes https://github.com/artsy/force/issues/86

### CODE REVIEW

**Comment by broskoski** [comment]
@kanaabe @1aurabrown is out, I'll take a look!

File: apps/partner/client/articles.coffee [hunk 1]
```diff
@@ -1,12 +1,57 @@
...
-  constructor: ({ profile, partner, cache, el }) ->
-    collection = new Articles
-    collection.url = "#{collection.url}/?partner_id=#{partner.get('_id')}&published=true&limit=12&sort=-published_at"
-    view = new ArticlesGridView el: el, collection: collection
-    el.html '<div class="loading-spinner"></div>'
-    collection.fetch()
-    view
\ No newline at end of file
+  constructor: ({ @profile, @partner, @cache, @el }) ->
+    @renderArticle() if @isArticle()
```
Comment (broskoski) [inline apps/partner/client/articles.coffee]: I wonder if this is easier to read as a normal `if, else` statement #soveryminor

File: apps/partner/client/articles.coffee [hunk 2]
```diff
@@ -1,12 +1,57 @@
...
+    @renderArticlesGrid() if not @isArticle()
+
+  isArticle: ->
+    window.location.pathname.includes '/article/'
```
Comment (broskoski) [inline apps/partner/client/articles.coffee]: Its there a stronger way to evaluate this? Like checking for the presence of `sd.ARTICLE`?

File: apps/partner/client/articles.coffee [hunk 3]
```diff
@@ -1,12 +1,57 @@
...
+
+  renderArticlesGrid: ->
+    @collection = new Articles
+    @collection.url = "#{@collection.url}/?partner_id=#{@partner.get('_id')}&published=true&limit=12&sort=-published_at"
```
Comment (broskoski) [inline apps/partner/client/articles.coffee]: Always wonder if there are better ways when I see these interpolated urls attached to collections. Maybe pass a `params` object to `ArticlesGridView`?

File: apps/partner/client/articles.coffee [hunk 4]
```diff
@@ -1,12 +1,57 @@
...
+    view = new ArticlesGridView
+      el: @el
+      collection: @collection
+      partner: @partner
+    @el.html '<div class="loading-spinner"></div>'
+    @collection.fetch()
+    view
+
+  renderArticle: ->
+    slug = _.last window.location.pathname.split('/')
```
Comment (broskoski) [inline apps/partner/client/articles.coffee]: Again, this feels like there's room for this getting fucked up in the future. Maybe pulling the slug off of `sd.ARTICLE` is better (but could be missing part of the flow here).

**Comment by broskoski** [comment]
Works for me! Comments above are more geared towards refactoring than hard blockers. Feel free to merge when ready.

**Comment by kanaabe** [comment]
Thanks @broskoski -- I'm going to merge for now and think a little more on how to get the `sd.ARTICLE` or something similar in. The routing logic initializes the section's views [here](https://github.com/artsy/force/blob/master/apps/partner/client/view.coffee#L55) but we don't fetch the article beforehand, so I _think_ it's tricky (unless I'm missing something obvious). I do agree that it should be changed.

**Pull request merged**